src/seedu/addressbook/data/person/Address.java

```diff
@@ -26,6 +38,15 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
         this.value = address;
```

Comment:
Well, that's my initial thought too. However, it is still necessary to perform this check before parsing (since it can't happen if it's malformed).